src/modules/cl/kernel/crop_mirror_normalize.cl

```diff
@@ -1,169 +1,373 @@
 #pragma OPENCL EXTENSION cl_amd_media_ops : enable
 #pragma OPENCL EXTENSION cl_amd_media_ops2 : enable
-#define saturate_8u(value) ( (value) > 255 ? 255 : ((value) < 0 ? 0 : (value) ))
+#pragma OPENCL EXTENSION cl_khr_fp16 : enable
+#define saturate_8u(value) ((value) > 255 ? 255 : ((value) < 0 ? 0 : (value)))
 
+kernel void crop_mirror_normalize_batch(
+    __global unsigned char *input, 
+    __global unsigned char *output, 
+    __global unsigned int *dst_height, __global unsigned int *dst_width,
+    __global unsigned int *src_width, __global unsigned int *start_x,
+    __global unsigned int *start_y, __global float *mean,
+    __global float *std_dev, __global unsigned int *flip,
+    __global unsigned int *max_src_width, __global unsigned int *max_dst_width,
+    __global unsigned long *src_batch_index,
+    __global unsigned long *dst_batch_index, const unsigned int channel,
+    // const unsigned int batch_size,
+    __global unsigned int *src_inc,
+    __global unsigned int *dst_inc,
+    const int in_plnpkdind, const int out_plnpkdind // use 1 pln 3 for pkd
```

Comment:
consider doing more work in the kernel and use vector data types for better performance